File: Sources/Partout/Partout.swift
// SPDX-FileCopyrightText: 2025 Davide De Rosa
//
// SPDX-License-Identifier: GPL-3.0

import Foundation

// MARK: Core

@_exported import PartoutCore
@_exported import PartoutProviders

// MARK: - Optional

#if canImport(PartoutOpenVPN)
@_exported import PartoutOpenVPN
#endif

#if canImport(PartoutWireGuard)
@_exported import PartoutWireGuard
#endif

#if canImport(PartoutAPI)
@_exported import PartoutAPI
@_exported import PartoutAPIBundle
#endif

// MARK: - Vendors

@_exported import _PartoutVendorsPortable
#if canImport(_PartoutVendorsApple)
@_exported import _PartoutVendorsApple
@_exported import _PartoutVendorsAppleNE
#endif

File: cross/apple/Sources/Partout_C/partout.c
/*
 * SPDX-FileCopyrightText: 2026 Davide De Rosa
 *
 * SPDX-License-Identifier: GPL-3.0
 */

#include "partout.h"
#include "portable/portable.h"

#if PARTOUT_HAS_CRYPTO
#include "crypto/crypto.h"
#endif

#if PARTOUT_OPENVPN
#include "openvpn/openvpn.h"
#endif
#if PARTOUT_WIREGUARD
#include "wireguard/backend.h"
#include "wireguard/wireguard.h"
#endif
